src/util/RtreeBuild.cpp

```diff
@@ -0,0 +1,1133 @@
+        splitResult.bestElement);
+    splitBuffers.splitBuffersRam.s1SmallDim1->push_back(
+        splitResult.bestMaxElement);
+
+    // placeholder
+    otherDimension.smallSplit0 = splitBuffers.splitBuffersRam.s0SmallDim0;
+    otherDimension.smallSplit1 = splitBuffers.splitBuffersRam.s1SmallDim0;
+
+    otherDimension.smallSplit0->push_back(RTreeValueWithOrderIndex());
+    otherDimension.smallSplit0->push_back(RTreeValueWithOrderIndex());
+    otherDimension.smallSplit1->push_back(RTreeValueWithOrderIndex());
+    otherDimension.smallSplit1->push_back(RTreeValueWithOrderIndex());
+  }
+
+  std::optional<RTreeValueWithOrderIndex> elementOpt;
+  std::optional<FileReader> fileReaderDim0;
+  std::optional<FileReader> fileReaderDim1;
+  if (!this->workInRam) {
+    fileReaderDim0 = {FileReader(this->rectsD0.rectanglesOnDisk)};
+    fileReaderDim1 = {FileReader(this->rectsD1.rectanglesOnDisk)};
+  }
+  uint64_t currentXSplit0 = 0;
+  uint64_t currentXSplit1 = 0;
+  uint64_t currentYSplit0 = 0;
+  uint64_t currentYSplit1 = 0;
+  for (size_t dim = 0; dim < 2; dim++) {
+    // start performing the actual split
+    uint64_t i = 0;
+
+    if (!this->workInRam) {
+      if (dim == 0) elementOpt = fileReaderDim0.value().GetNextElement();
+      if (dim == 1) elementOpt = fileReaderDim1.value().GetNextElement();
+    }
+
+    while ((this->workInRam && i < this->size) ||
+           (!this->workInRam && elementOpt)) {
+      RTreeValueWithOrderIndex element;
+
+      // get the current element, either from disk or from ram
+      if (this->workInRam) {
+        element = dim == 0 ? (*this->rectsD0.rectanglesInRam)[i]
+                           : (*this->rectsD1.rectanglesInRam)[i];
+      } else {
+        element = elementOpt.value();
+      }
+
+      if ((splitResult.bestDim == 0 &&
+           element.orderX < splitResult.bestElement.orderX) ||
+          (splitResult.bestDim == 1 &&
+           element.orderY < splitResult.bestElement.orderY)) {
+        // the element belongs to split 0
+
+        if (dim == 0) {
+          // add the element to the split 0 dimension 0 vector / file
+          if (split0InRam || this->workInRam) {
+            splitBuffers.splitBuffersRam.s0Dim0->push_back(element);
+          } else {
+            Rtree::SaveEntryWithOrderIndex(element,
+                                           splitBuffers.split0Dim0File.value());
+          }
+
+          // check if the element is at the position i * S (described in the
+          // algorithm) or one before it. In this case it is a future possible
+          // split position and needs to be saved to the "small list"
+          if (((currentXSplit0 + 1) % SSplit0 == 0 &&
+               (currentXSplit0 + 1) / SSplit0 >= 1 &&
+               (currentXSplit0 + 1) / SSplit0 < M) ||
+              (currentXSplit0 % SSplit0 == 0 && currentXSplit0 / SSplit0 >= 1 &&
+               currentXSplit0 / SSplit0 < M)) {
+            // index i * S - 1 or i * S
```

Comment:
This check (or similar ones) appear very often. 
Factor this out into a function `isBorderOfSplitCandidate(...)`